clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -99,25 +86,138 @@ public Coordinator(ConsumerNetworkClient client,
                        OffsetCommitCallback defaultOffsetCommitCallback,
                        boolean autoCommitEnabled,
                        long autoCommitIntervalMs) {
-        this.client = client;
-        this.time = time;
-        this.generation = -1;
-        this.consumerId = JoinGroupRequest.UNKNOWN_CONSUMER_ID;
-        this.groupId = groupId;
-        this.consumerCoordinator = null;
+        super(new ConsumerProtocol(),
+                client,
+                groupId,
+                sessionTimeoutMs,
+                heartbeatIntervalMs,
+                metrics,
+                metricGrpPrefix,
+                metricTags,
+                time,
+                requestTimeoutMs,
+                retryBackoffMs);
+        this.metadata = metadata;
+        this.metadata.addListener(this);
+        this.metadata.requestUpdate();
+        this.metadataSnapshot = new MetadataSnapshot();
         this.subscriptions = subscriptions;
-        this.sessionTimeoutMs = sessionTimeoutMs;
-        this.assignmentStrategy = assignmentStrategy;
-        this.heartbeat = new Heartbeat(this.sessionTimeoutMs, heartbeatIntervalMs, time.milliseconds());
-        this.heartbeatTask = new HeartbeatTask();
-        this.sensors = new CoordinatorMetrics(metrics, metricGrpPrefix, metricTags);
-        this.requestTimeoutMs = requestTimeoutMs;
-        this.retryBackoffMs = retryBackoffMs;
         this.defaultOffsetCommitCallback = defaultOffsetCommitCallback;
         this.autoCommitEnabled = autoCommitEnabled;
 
+        this.assignors = new HashMap<>();
+        for (PartitionAssignor assignor : assignors)
+            this.assignors.put(assignor.name(), assignor);
```

Comment:
nit: rename assignors to assignorsMap? it's a bit misleading to have two variables with the same name here.